components/filter/filter-dimension-set-date-text-value.js

```diff
@@ -0,0 +1,156 @@
+ * A range property is used to define the preset text shown, as well as the start and end date values formatted as UTC strings.
+ * This component does not render anything, but instead gathers data needed for the d2l-filter.
+ */
+class FilterDimensionSetDateTextValue extends LocalizeCoreElement(LitElement) {
+
+	static get properties() {
+		return {
+			/**
+			 * Whether this value in the filter is disabled or not
+			 * @type {boolean}
+			 */
+			disabled: { type: Boolean, reflect: true },
+			/**
+			 * @ignore
+			 */
+			endValue: { type: String },
+			/**
+			 * REQUIRED: Unique key to represent this value in the dimension
+			 * @type {string}
+			 */
+			key: { type: String },
+			/**
+			 * REQUIRED: The preset date/time range that the list item represents
+			 * @type {'today'|'lastHour'|'24hours'|'48hours'|'7days'|'14days'|'30days'|'6months'}
+			 */
+			range: { type: String },
+			/**
+			 * Whether this value in the filter is selected or not
+			 * @type {boolean}
+			 */
+			selected: { type: Boolean, reflect: true },
+			/**
+			 * @ignore
+			 */
+			startValue: { type: String },
+			/**
+			 * @ignore
+			 */
+			text: { type: String, reflect: true }
+		};
+	}
+
+	constructor() {
+		super();
+		this.disabled = false;
+		this.selected = false;
+		this.text = '';
+		this._enforceSingleSelection = true;
+		this._filterSetValue = true;
+		this._noSearchSupport = true;
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		if (this.selected && this.range) {
+			// if the value is initially selected, startValue and endValue should be set in case used by consumer
+			const dateTimeRange = getUTCDateTimeRange(this._rangeDetails?.rangeType, this._rangeDetails?.rangeNum);
+			this.startValue = dateTimeRange?.startValue;
+			this.endValue = dateTimeRange?.endValue;
+		}
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+
+		const changes = new Map();
+		changedProperties.forEach((oldValue, prop) => {
+			if (oldValue === undefined && prop !== 'text') return;
+
+			if (prop === 'disabled' || prop === 'selected' || prop === 'text') {
+				changes.set(prop, this[prop]);
+			} else if (prop === 'range') {
+				changes.set('rangeType', this._rangeDetails?.rangeType);
+				changes.set('rangeNum', this._rangeDetails?.rangeNum);
+			}
+		});
+		if (changes.size > 0) {
+			/** @ignore */
+			this.dispatchEvent(new CustomEvent('d2l-filter-dimension-set-value-data-change', {
+				detail: { valueKey: this.key, changes: changes },
+				bubbles: true,
+				composed: false
+			}));
+		}
+	}
+
+	willUpdate(changedProperties) {
+		super.willUpdate(changedProperties);
+
+		if (changedProperties.has('range')) this._handleRangeUpdated();
+	}
+
+	getValueDetails() {
+		this._handleRangeUpdated();
+
+		return {
+			disabled: this.disabled,
+			key: this.key,
+			selected: this.selected,
+			text: this.text,
+			getAdditionalEventDetails: this._getAdditionalEventDetails.bind(this)
```

Comment:
I guess we can't just provide `startValue` and `endValue` here eh (i.e. instead of a method)?

Reply:
We want `startValue` and `endValue` to be up to date as of when the user selects them. If we set them here, they would only be up to date when `getValueDetails` runs, which currently is only on slot change. These in theory could be fairly different times depending on the length of time between the slot change and selection.